Nodejs/Product/TargetsVsix/Microsoft.NodejsToolsV2.targets

```diff
@@ -0,0 +1,313 @@
+  <Target Name="GetTargetFrameworkMonikerDisplayName" Returns="" />
+  <Target Name="CreateManifestResourceNames" Returns=""/>
+
+  <Import Project="$(MSBuildToolsPath)\Microsoft.Common.targets"/>
+  <Import Project="$(VSToolsPath)\Docker Tools\Microsoft.DockerTools.Publish.targets" Condition="Exists('$(VSToolsPath)\Docker Tools\Microsoft.DockerTools.Publish.targets')"/>
+
+  <Target
+    Name="GetReferenceAssemblyPaths"
+    DependsOnTargets="$(GetReferenceAssemblyPathsDependsOn)">
+    <!-- We have no reference assembly paths -->
+  </Target>
+
+  <ItemGroup>
+    <NodeModulesContent Include="$(ProjectHome)\node_modules\**\*.*"/>
+  </ItemGroup>
+
+  <Target Name="CopyFilesToOutputDirectory"/>
+
+  <Target Name="AfterClean">
+    <Exec Condition="'$(EnableTypeScript)' == 'true'" Command="npm run clean" />
+  </Target>
+
+  <Target Name="AfterCompile">
+    <Exec Condition="'$(EnableTypeScript)' == 'true'" Command="npm run build" />
+  </Target>
+
+  <Target Name="CoreCompile">
+    <Error Condition="'$(EnableTypeScript)' == 'true' And !Exists('$(QualifiedProjectHome)node_modules\.bin\tsc')" Text="TypeScript compiler is not installed. Please execute 'npm install --save-dev typescript'"></Error>
+    <PropertyGroup>
+      <_NodeModulesContentProperty>@(NodeModulesContent)</_NodeModulesContentProperty>
+    </PropertyGroup>
+    <Error Condition="$(_NodeModulesContentProperty.IndexOf('*')) != -1"
+           Text="Files in your project exceed MAX_PATH, please move your project to a location with a shorter path."></Error>
+
+    <!-- Target used for generating a package which can be uploaded to web server -->
+    <ItemGroup>
+      <_FoundWebConfig Include="@(Content)" Condition=" '%(Content.Filename)%(Content.Extension)' == 'web.config' " />
+      <_FoundWebConfig Include="@(Include)" Condition=" '%(Include.Filename)%(Include.Extension)' == 'web.config' " />
+    </ItemGroup>
+
+    <Copy
+      SourceFiles="@(NodejsAzureDeploymentFile)"
+      DestinationFolder="bin"
+      SkipUnchangedFiles="true">
+      <Output TaskParameter="DestinationFiles" ItemName="FileWrites"/>
+      <Output TaskParameter="DestinationFiles" ItemName="FilesCopiedToIntermediate"/>
+    </Copy>
+  </Target>
+
+  <Target Name="BuiltProjectOutputGroup"
+          Returns="@(BuiltProjectOutputGroupOutput)"
+          DependsOnTargets="$(BuiltProjectOutputGroupDependsOn)">
+    <ItemGroup>
+      <!-- We can't depend upon GeneratedJavascript defined by TypeScript because it 
+           doesn't occur when we're publishing, just when building.  So we calculate
+           the paths for the generated code here.  We also don't get to rely upon 
+           TypeScriptCollectPublishFiles to include the files for publishing because 
+           Worker Roles won't invoke PublishPipeline, they invoke some targets directly -->
+      <_GeneratedJavascript Include="@(TypeScriptCompile->'%(RelativeDir)%(Filename).js');@(Compile->'%(RelativeDir)%(Filename).js')"
```

Comment:
I'm surprised the file extension is hard-coded.  There's not a variable for that?  What about jsx files?

Reply:
There's no variable. `jsx` and `tsx` files are transpiled to `js` eventually. Note that this will be configured by `tsconfig.json`, meaning that `jsx` will not be transpiled unless `AllowJs=true` in the `tsconfig.json` file.

Reply:
What about `%(Extension)`?

---
In reply to: [249956829](https://github.com/Microsoft/nodejstools/pull/2123#discussion_r249956829) [](ancestors = 249956829)